cmd/utils/vmss.go

```diff
@@ -78,6 +80,30 @@ func VirtualMachineScaleSetVMFromConfig() (*VirtualMachineScaleSetVM, error) {
 	return &vm, nil
 }
 
+func VirtualMachineScaleSetVMsViaKubeconfig() (map[string]*VirtualMachineScaleSetVM, error) {
+	clientset, err := k8sutil.NewClientsetFromConfigFlags(KubernetesConfigFlags)
+	if err != nil {
+		return nil, fmt.Errorf("creating Kubernetes client: %w", err)
+	}
+
+	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metaV1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("listing nodes: %w", err)
+	}
+
+	var vmssVMs = make(map[string]*VirtualMachineScaleSetVM)
+	if len(nodes.Items) > 0 {
+		for _, n := range nodes.Items {
+			var vm VirtualMachineScaleSetVM
+			if err = ParseVMSSResourceID(strings.TrimPrefix(n.Spec.ProviderID, "azure://"), &vm); err != nil {
+				return nil, fmt.Errorf("parsing Azure resource ID %s: %w", n.Spec.ProviderID, err)
```

Comment:
```suggestion
				return nil, fmt.Errorf("parsing Azure resource ID %q: %w", n.Spec.ProviderID, err)
```

I tried it with a non-AKS cluster and this was the error:

```bash
$ ./kubectl-az config import
Error: failed to get VMSS VMs: parsing Azure resource ID : error parsing provider ID : unexpected EOF
```

Please apply the same change in `ParseVMSSResourceID`: 

https://github.com/Azure/kubectl-az/blob/53c3a54530a98d5378a0e66acba2dbaaa99255a4/cmd/utils/vmss.go#L44